File: Baekjoon/16000~16999/[16969]차량 번호판 2/solution.cpp
#include <iostream>
#include <string>

using namespace std;

#define endl "\n"
#define MOD 1'000'000'009

string n_format;
long long res;
bool flag;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  cin >> n_format;
  res = 1;

  for (int i = 0; i < n_format.length(); i++) {
    if (i == 0) {
      if (n_format[i] == 'c') {
        flag = false;
        res = (res * 26) % MOD;
      } else {
        flag = true;
        res = (res * 10) % MOD;
      }
      continue;
    }

    if (n_format[i] == 'c') {
      res = (res * (flag ? 26 : 25)) % MOD;
      flag = false;
    } else {
      res = (res * (flag ? 9 : 10)) % MOD;
      flag = true;
    }
  }

  cout << res << endl;

  return 0;
}

File: Baekjoon/16000~16999/[16969]차량 번호판 2/solution.py
import sys

n_format = sys.stdin.readline()
res = 1
flag = 0
MOD = 1_000_000_009

for c in n_format:
    if c == 'c':
        res = (res * (25 if flag == 1 else 26)) % MOD
        flag = 1
    elif c == 'd':
        res = (res * (9 if flag == 2 else 10)) % MOD
        flag = 2

sys.stdout.write(str(res) + "\n")
